Index: python/BTaggingConfiguration_LoadTools.py
===================================================================
--- python/BTaggingConfiguration_LoadTools.py	(revision 769371)
+++ python/BTaggingConfiguration_LoadTools.py	(working copy)
@@ -477,7 +477,8 @@
   if 'MV2c10hpFlip' in TaggerList:
     tag = ConfInstance.addTool('MV2c10hpFlipTag', ToolSvc, 'BTagTrackToJetAssociator', JetCollection, Verbose = BTaggingFlags.OutputLevel < 3)
     obj.MVTagToolList += [tag,]
-  if 'JetVertexCharge' in TaggerList:
+  if 'JetVertexChargeX' in TaggerList:
+    raise ValueError('wtf')
     ConfInstance.addTool('JetVertexCharge', ToolSvc, 'BTagTrackToJetAssociator', JetCollection, Verbose = BTaggingFlags.OutputLevel < 3)
 
   if 'DL1' in TaggerList:
@@ -685,8 +686,9 @@
     obj.MVTagToolList += [tag,]
     #ConfInstance.addTool('MV2mFlipTag', ToolSvc, 'BTagTrackToJetAssociator', JetCollection, Verbose = BTaggingFlags.OutputLevel < 3,
     #                     CheckOnlyInsideToolCollection=True, DoNotSetUpParticleAssociators=True)
-  if 'JetVertexCharge' in TaggerList:   #LC FIXME
+  if 'JetVertexChargeX' in TaggerList:   #LC FIXME
     #ConfInstance.setupMuonAssociator('Muons', JetCollection, ToolSvc, Verbose = BTaggingFlags.OutputLevel < 3)
+    raise ValueError('wtf')
     ConfInstance.addTool('JetVertexCharge', ToolSvc, 'BTagTrackToJetAssociator', JetCollection, Verbose = BTaggingFlags.OutputLevel < 3,
                          CheckOnlyInsideToolCollection=True, DoNotSetUpParticleAssociators=True)
   if 'DL1' in TaggerList:
@@ -796,7 +798,8 @@
     ConfInstance.addTool('MV1cFlipTag', ToolSvc, 'BTagTrackToJetAssociator', JetCollection, Verbose = BTaggingFlags.OutputLevel < 3)
   if 'MV1Flip' in TaggerList:
     ConfInstance.addTool('MV1FlipTag', ToolSvc, 'BTagTrackToJetAssociator', JetCollection, Verbose = BTaggingFlags.OutputLevel < 3)
-  if 'JetVertexCharge' in TaggerList:   #LC FIXME
+  if 'JetVertexChargeX' in TaggerList:   #LC FIXME
+    raise ValueError('wtf')
     ConfInstance.addTool('JetVertexCharge', ToolSvc, 'BTagTrackToJetAssociator', JetCollection, Verbose = BTaggingFlags.OutputLevel < 3)
   #MultivariateTagManager Taggers
 
