Index: FTAG5.py
===================================================================
--- FTAG5.py	(revision 772334)
+++ FTAG5.py	(working copy)
@@ -259,7 +259,7 @@
 
 else:
     triggers = []
-    expression = offlineExpression
+    expression = ''
 
 from DerivationFrameworkTools.DerivationFrameworkToolsConf import DerivationFramework__xAODStringSkimmingTool
 FTAG5SkimmingTool = DerivationFramework__xAODStringSkimmingTool(name = "FTAG5OfflineSkimmingTool", expression = expression)
